src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -367,69 +403,101 @@ export function useSwapInputsController({
     const params = buildQuoteParams({
       currentAddress: store.getState().settings.accountAddress,
       inputAmount,
-      outputAmount,
       inputAsset: internalSelectedInputAsset.value,
+      lastTypedInput: lastTypedInputParam,
+      outputAmount,
       outputAsset: internalSelectedOutputAsset.value,
-      lastTypedInput,
     });
 
     logger.debug(`[useSwapInputsController]: quote params`, {
       data: params,
     });
 
     if (!params) {
-      runOnUI(resetFetchingStatus)(true);
+      runOnUI(resetFetchingStatus)({ fromError: true });
       return;
     }
 
-    const response = (params.swapType === SwapType.crossChain ? await getCrosschainQuote(params) : await getQuote(params)) as
-      | Quote
-      | CrosschainQuote
-      | QuoteError;
-
-    logger.debug(`[useSwapInputsController]: quote response`, {
-      data: response,
-    });
-
-    // TODO: Handle native asset inputs
-    runOnUI(setQuote)({
-      data: response,
-      outputAmount:
-        lastTypedInput === 'inputAmount'
+    try {
+      const [quoteResponse, fetchedPrices] = await Promise.all([
+        params.swapType === SwapType.crossChain ? getCrosschainQuote(params) : getQuote(params),
+        fetchAssetPrices({
+          inputAsset: internalSelectedInputAsset.value,
+          outputAsset: internalSelectedOutputAsset.value,
+        }),
+      ]);
+
+      const quotedInputAmount =
+        lastTypedInputParam === 'outputAmount'
           ? Number(
               convertRawAmountToDecimalFormat(
-                (response as Quote)?.buyAmountMinusFees?.toString(),
-                internalSelectedOutputAsset.value?.decimals || 18
+                (quoteResponse as Quote)?.sellAmount?.toString(),
+                internalSelectedInputAsset.value?.decimals || 18
               )
             )
-          : undefined,
-      inputAmount:
-        lastTypedInput === 'outputAmount'
+          : undefined;
+
+      const quotedOutputAmount =
+        lastTypedInputParam === 'inputAmount'
           ? Number(
-              convertRawAmountToDecimalFormat((response as Quote)?.sellAmount?.toString(), internalSelectedInputAsset.value?.decimals || 18)
+              convertRawAmountToDecimalFormat(
+                (quoteResponse as Quote)?.buyAmountMinusFees?.toString(),
+                internalSelectedOutputAsset.value?.decimals || 18
+              )
             )
-          : undefined,
-    });
-    runOnUI(resetFetchingStatus)((response as QuoteError)?.error);
+          : undefined;
+
+      // TODO: Handle native asset inputs
+      runOnUI(() => {
+        // Check whether the quote has been superseded by new user input so we don't introduce conflicting updates
+        const isLastTypedInputStillValid = lastTypedInputParam === lastTypedInput.value;
```

Comment:
this logic below for checking of the quote is still valid should be the responsibility of the `setQuote` worklet which should be passed the original quote params to compare the current state against.

`setQuote` could be renamed to something like `validateAndSetQuoteWorklet` or `parseAndSetQuoteWorklet` etc
    
below, we are taking these from the params (passed into `buildQuoteParams`)
- lastTypedInputParam
- inputAmount
- outputAmount

and comparing it to the live state of:
- lastTypedInput.value
- inputValues.value.inputAmount
- inputValues.value.outputAmount

If we want to be thorough, we should compare it to the full params of the quote request.
Basically, we want to base it off of the params to `buildQuoteParams`
Edge case scenario: `lastTypedInputParam` is `inputAmount` and the value is `5`; we switch assets but the input amount is still the same